modules/ROOT/pages/deployment/extensions/ocdav.adoc

```diff
@@ -1,18 +1,62 @@
 = OCDAV Extension Configuration
 :toc: right
+:grpc-url: https://grpc.io/about/
+:cs3-url: http://www.cs3community.org
 
 :ext_name: ocdav
 
-:description: The Infinite Scale OCDAV extension provides ...
+:description: The ocDAV extension is responsible for translating ownCloud flavoured WebDAV into CS3 API calls.
+
+// references https://github.com/owncloud/ocis/pull/3864 ([docs-only] add PROPFIND sequence diagram examples)
+// also see: https://owncloud.dev/architecture/protocol-changes/ 
 
 == Introduction
 
 {description}
 
+{cs3-url}[CS3] (Cloud Storage Services for Synchronization and Sharing) is based on {grpc-url}[GRPC] (open source high performance Remote Procedure Call (RPC) framework) which uses a binary coded versionable payload protocol with much higher efficiency when it comes to parsing compared to a classical xml payload.
+
 // fixme: source description is still missing, and a better description is needed
 
 include::partial$deployment/extensions/beta-statement.adoc[]
 
+== Sequence Diagram
+
+// https://github.com/owncloud/ocis/pull/3864
+// https://owncloud.dev/architecture/protocol-changes/
+ 
+=== General View
+
+A PROPFIND finds its way to a storage provider like this. While this is a simplification to get an understanding of what needs to go where, there are several places where sharding can happen.
+
+image::deployment/extensions/ocdav/mermaid-ocdav-1.svg[width=600]
+
+=== Proxy Based User Routing
+
+The Infinite Scale proxy authenticates requests an can forward requests to different backends, depending on the logged in user or cookies. For example multiple ocdav services can be configured to shard users based on username or affiliation.
+
+image::deployment/extensions/ocdav/mermaid-ocdav-2.svg[width=600]
+
+=== Gateway Based Path or Storage Provider Id Based Routing
+
+The reva gateway acts as a facade to multiple storage providers that can be configured with the storage registry.
+
+image::deployment/extensions/ocdav/mermaid-ocdav-3.svg[width=600]
+
+=== PROPFIND Request Against Old Webdav Endpoints
+
+Old `/dav/files/\{username}` endpoint with username and a path relative to the users home.
+
+To route a PROPFIND request against the old webdav endpoints like `/dav/files/username`, ocdav first has to build a CS3 namespace prefix, eg. `/users/{{.Id.OpaqueId}}` to the users home.
+
+image::deployment/extensions/ocdav/mermaid-ocdav-4.svg[width=600]
+
+=== Handling Legacy Global Namespace Webdav Endpoints
+
+The reason Infinite Scale uses a path based lookup instead of looking up the current users home using the user id and a space type filter is, because there are deployments that use a global namespace at the legacy `/webdav` endpoint. To support these use cases, the gateway allows looking up spaces using their mount path.
```

Comment:
```suggestion
The reason Infinite Scale uses a path-based lookup instead of looking up the current user's home using the user ID and a space type filter is, because there are deployments that use a global namespace at the legacy `/webdav` endpoint. To support these use cases, the gateway allows looking up spaces using their mount path.
```